openpathsampling/snapshot.py

```diff
@@ -313,22 +258,137 @@ def inner(self, *args, **kwargs):
+               vectors, etc.
+        If you want to obtain a snapshot from a currently-running MD engine,
+        use that engine's .current_snapshot property.
+
+        Parameters
+        ----------
+        coordinates : simtk.unit.Quantity wrapping Nx3 np array of dimension length
+            atomic coordinates (default: None)
+        velocities : simtk.unit.Quantity wrapping Nx3 np array of dimension length
+            atomic velocities (default: None)
+        box_vectors : periodic box vectors (default: None)
+            the periodic box vectors at current timestep (defautl: None)
+        potential_energy : simtk.unit.Quantity of units energy/mole
+            potential energy at current timestep (default: None)
+        kinetic_energy : simtk.unit.Quantity of units energy/mole
+            kinetic energy at current timestep (default: None)
+
+        Attributes
+        ----------
+        coordinates : simtk.unit.Quantity wrapping Nx3 np array of dimension length
+            atomic coordinates
+        velocities : simtk.unit.Quantity wrapping Nx3 np array of dimension length
+            atomic velocities
+        box_vectors : periodic box vectors
+            the periodic box vectors
+        potential_energy : simtk.unit.Quantity of units energy/mole
+            potential energy
+        kinetic_energy : simtk.unit.Quantity of units energy/mole
+            kinetic energy
+        idx : dict( Storage() : int )
+            dict for storing the used index per storage
+        """
+
+        super(AbstractSnapshot, self).__init__()
+
+        self.is_reversed = is_reversed
+        self.topology = topology
+
+        if reversed_copy is None:
+            # this will always create the mirrored copy so we can save in pairs!
+            self._reversed = self.__class__.__new__(self.__class__)
+            AbstractSnapshot.__init__(
+                self._reversed,
+                is_reversed=not self.is_reversed,
+                reversed_copy=self,
+                topology=topology
+            )
+
+        else:
+            self._reversed = reversed_copy
+
+    def __eq__(self, other):
+        if self is other:
+            return True
+        elif hasattr(other, '_idx'):
+            if other.__subject__ is self:
+                return True
+
+        return False
+
+    # ==========================================================================
+    # Utility functions
+    # ==========================================================================
+
+    def copy(self):
+        """
+        Returns a shallow copy of the instance itself. The contained
+        configuration and momenta are not copied.
+
+        This will also lead to a new reversed copy when using reversed!
+
+        Returns
+        -------
+        Snapshot()
+            the deep copy
+        """
+        this = AbstractSnapshot(is_reversed=self.is_reversed, topology=self.topology)
```

Comment:
Why does `AbstractSnapshot` implement this? Seems kind of weird, since this snapshot shouldn't be possible to instantiate, if it is actually abstract.